stl/inc/vector

```diff
@@ -812,12 +1055,16 @@ private:
         const auto _Whereoff = static_cast<size_type>(_Where._Ptr - _Myfirst);
         const auto _Oldsize  = static_cast<size_type>(_Mylast - _Myfirst);
 

```

Comment:
One thing we can/should do to help reduce ODR issues is to change the name of any ugly function that has been updated to use `_ASAN_ANNOTATE` or that has a direct or indirect callee which use `_ASAN_ANNOTATE`. (Our conventional mechanism for such name changes is to add a `2` to the end of the name, or increment this "version" if the name already ends in a digit.) Such a name change avoids the "linker chooses old function instead of new one" ODR problem.

This overload and the `forward_iterator_tag` overload of `_Insert_range` (and the similar overloads of `_Assign_range`, and `_Resize`) are candidates, for example. (I'll let you audit for "any ugly functions that call them, perhaps indirectly".)

Reply:
Won't help if `_Ugly` is inlined into `non_ugly`, and `non_ugly` is not inlined, which is likely enough to happen

Reply:
Hence my claim this can "reduce ODR issues" and description of one scenario (possibly the only scenario) that it fixes.